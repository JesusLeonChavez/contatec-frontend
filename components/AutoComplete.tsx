import {
  Input,
  Box,
  InputGroup,
  InputLeftElement,
  FormControl,
  FormLabel,
  Button,
  Flex
} from "@chakra-ui/react"
import router from "next/router"
import { useRef } from "react"
import { connectAutoComplete } from "react-instantsearch-dom"
import ZIcon from "../components/Icon"
import { post } from "../utils/http"
import showToast from "./Toast"

// Aplicar  estilos
export const Autocomplete = ({ hits, currentRefinement, refine }) => {
  const inputRef = useRef<HTMLInputElement>(null)
  const handleSelect = hit => {
    console.log(hit)
    refine(
      `${hit.description.charAt(0).toUpperCase()}${hit.description
        .split("_")
        .join(" ")
        .slice(1)}`
    )
    // eslint-disable-next-line camelcase
    const { objectID, category_id } = hit
    setTimeout(() => {
      // eslint-disable-next-line camelcase
      router.push(`/explorar/${category_id}/${objectID}`)
    }, 500)
  }
  const handleSubmit = async e => {
    e.preventDefault()
    const res = await post("/api/post/search", {
      nombre_post: inputRef.current?.value,
      categoria_post: inputRef.current?.value
    })
    if (res.data.message === "Http Exception")
      return showToast(
        "Error",
        "No se encontraron servicios relacionados a su busqueda",
        "error"
      )
    const { idCategoria, idPost } = res.data.data
    router.push(`/explorar/${idCategoria}/${idPost}`)
    // console.log("api/buscar/" + inputRef.current?.value || "none")
  }

  return (
    <form onSubmit={handleSubmit}>
      <FormControl id="text">
        <FormLabel>Busca el tema de tu inter√©s</FormLabel>
        <Flex>
          <ul style={{ textDecoration: "none", listStyle: "none" }}>
            <li>
              <InputGroup w={{ base: "50", md: "md", lg: "lg" }}>
                <InputLeftElement
                  pointerEvents="none"
                  // eslint-disable-next-line react/no-children-prop
                  children={<ZIcon name="search" />}
                />
                <Input
                  ref={inputRef}
                  type="search"
                  placeholder="P. ej: Desarrollo web"
                  bg="white"
                  name="search"
                  value={currentRefinement}
                  onChange={event => refine(event.currentTarget.value)}
                  autoComplete="off"
                />
              </InputGroup>
            </li>
            {currentRefinement &&
              hits.map(hit => (
                <Box
                  key={hit.objectID}
                  bg="white"
                  color="primary"
                  p="3"
                  cursor="pointer"
                  transition="all 0.1s ease-in-out"
                  shadow="xl"
                  _hover={{ backgroundColor: "primary", color: "white" }}
                  onClick={() => {
                    handleSelect(hit)
                  }}
                >
                  {hit.description.charAt(0).toUpperCase()}
                  {hit.description.split("_").join(" ").slice(1)}
                </Box>
              ))}
          </ul>
          <Button w={{ md: "3xs" }} variant="primary" type="submit">
            Buscar
          </Button>
        </Flex>
      </FormControl>
    </form>
  )
}

export const CustomAutocomplete = connectAutoComplete(Autocomplete)
