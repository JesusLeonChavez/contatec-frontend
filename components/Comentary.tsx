import { Box, Flex, Text, Avatar } from "@chakra-ui/react"
import ZIcon from "./Icon/ZIcon"

interface ComentaryProps {
  img: string
  name: string
  comentary: string
  quantityStars: number
}
export default function Comentary({
  img,
  name,
  comentary,
  quantityStars
}: ComentaryProps) {
  return (
    <Box borderBottomWidth="1px" py="6">
      <Flex align="center" justify="start" py="6">
        <Avatar size="lg" name={name} src={img} mx="1" alt={name} />
        <Box mx="3">
          <Text>{name}</Text>
          <Flex align="center">
            <Text>{quantityStars}</Text>
            <ZIcon name="star" color="secondary" />
          </Flex>
        </Box>
      </Flex>
      <Box>{comentary}</Box>
    </Box>
  )
}
