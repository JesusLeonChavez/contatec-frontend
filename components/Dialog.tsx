import React, { useRef } from "react"
import {
  AlertDialog,
  AlertDialogBody,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogContent,
  AlertDialogOverlay,
  useDisclosure,
  Button,
  Text
} from "@chakra-ui/react"

import ZIcon from "../components/Icon"
import { IconName } from "./Icon/types"
type Color = "info" | "danger" | "primary" | "warning" | "success"

interface DialogProps {
  title: string
  content: React.ReactElement<any, any>
  accept?: string
  cancel?: string
  callbackFunction: (a: boolean) => void
  color: Color
  icon: IconName
}
export default function Dialog({
  title,
  content,
  accept = "SÃ­, acepto",
  cancel = "No, cancelar",
  callbackFunction,
  color = "primary",
  icon = "trash"
}: DialogProps) {
  const { onClose } = useDisclosure()
  const cancelRef = useRef(null)
  return (
    <>
      <AlertDialog
        motionPreset="slideInBottom"
        leastDestructiveRef={cancelRef}
        onClose={onClose}
        isOpen={true}
        isCentered
      >
        <AlertDialogOverlay />

        <AlertDialogContent py="4">
          <AlertDialogHeader
            d="flex"
            justifyContent="center"
            color={color}
            fontSize="2xl"
          >
            {title}
          </AlertDialogHeader>

          <AlertDialogBody
            d="flex"
            justifyContent="center"
            flexDirection="column"
            alignItems="center"
          >
            <ZIcon name={icon} size={50} color={color} />
            <Text align="center" p="4" color="primary">
              {content}
            </Text>
          </AlertDialogBody>

          <AlertDialogFooter d="flex" justifyContent="center">
            <Button
              variant="light"
              ref={cancelRef}
              onClick={() => {
                callbackFunction(false)
              }}
            >
              {cancel}
            </Button>

            <Button
              variant={color}
              ml={3}
              onClick={() => {
                callbackFunction(true)
              }}
            >
              {accept}
            </Button>
          </AlertDialogFooter>
        </AlertDialogContent>
      </AlertDialog>
    </>
  )
}
