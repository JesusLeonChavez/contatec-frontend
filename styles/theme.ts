import { extendTheme } from '@chakra-ui/react'
import { ButtonStyles as Button } from './components/buttonStyles'
export const newTheme = extendTheme({
  colors: {
    primary: '#EB0029',
    secondary: '#FF6F91',
    highlight: '#00C9A7'
  },
  components: {
    Button
  }
})
