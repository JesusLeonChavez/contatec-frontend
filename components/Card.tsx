import { Box, Text, Flex } from "@chakra-ui/react"
import Image from "next/image"
import styles from "../styles/sections/Home.module.css"
export default function Card({ title, price, img }) {
  // const property = {
  //   imageUrl: "https://bit.ly/2Z4KKcF",
  //   imageAlt: "Rear view of modern home with pool",
  //   beds: 3,
  //   baths: 2,
  //   title: "Modern home in city center in the heart of historic Los Angeles",
  //   formattedPrice: "$1,900.00",
  //   reviewCount: 34,
  //   rating: 4
  // }
  return (
    <Box borderWidth="1px" borderRadius="lg" overflow="hidden">
      <Image src={img} alt="Prueba img" height="300" width="300" />

      <Box p="5">
        <Flex align="center" justify="center" direction="column">
          <Flex align="center" justify="center">
            <Text fontSize="md" className={styles.bold500}>
              {title}
            </Text>
          </Flex>
          <Flex align="center" justify="center">
            <Text fontSize="md" className={styles.bold200} color="gray">
              Desde S/. {price}
            </Text>
          </Flex>
        </Flex>
      </Box>
    </Box>
  )
}
