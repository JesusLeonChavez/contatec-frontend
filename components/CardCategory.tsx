/* eslint-disable camelcase */
import { Box, Text, Flex } from "@chakra-ui/react"
import ZIcon from "../components/Icon/ZIcon"

import styles from "../styles/sections/Home.module.css"
import Image from "next/image"
import ModalNewPost from "../sections/Post/ModalNewPost"

interface User {
  avatar: string
  createdAt: string
  id: number
  updatedAt: string
  us_apellido: string
  us_correo: string
  us_nombre: string
}

interface Post {
  id: number
  createdAt: string
  updatedAt: string
  pst_isActive: boolean
  pst_descripcion_corta: string
  pst_nombre: string
  pst_descripcion_incluye: string
  pst_descripcion: string
  pst_imagen_1: string
  pst_imagen_2: string
  pst_imagen_3: string
  pst_imagen_4: string
  pst_imagen_5: string
  pst_precioBase: number
  pstUsuarioId: User
}

interface PropsCard {
  post: Post
  categoryScreen?: boolean
}
export default function CardCategory({
  post,
  categoryScreen = true
}: PropsCard) {
  return (
    <Box borderRadius="lg" overflow="hidden" mx="3">
      <Box position="relative">
        <Image
          src={post?.pst_imagen_5 || "/assets/images/marketing/marketing1.png"}
          alt={post?.pst_nombre}
          height="500"
          width="500"
        />
        <Flex
          align="center"
          justify="space-between"
          w="full"
          p="3"
          position="absolute"
          bottom="1"
          backgroundColor="whiteAlpha.700"
        >
          <Flex align="center" w="40px" justify="space-between">
            <ZIcon name="star" color="secondary" />
            <Text fontSize="sm" className={styles.bold400} color="primary">
              4.0
            </Text>
          </Flex>
          {categoryScreen ? (
            <Text fontSize="sm" className={styles.bold400} color="primary">
              S/. {post?.pst_precioBase}
            </Text>
          ) : (
            <ModalNewPost variant="primary" width="" icon mypost={post} />
          )}
        </Flex>
      </Box>

      <Box px="2" pb="5">
        <Flex align="flex-start" justify="center" direction="column">
          {/* <Flex align="center" justify="space-between" w="full" p="3">
            <Flex align="center" w="40px" justify="space-between">
              <ZIcon name="star" />
              <Text fontSize="sm" className={styles.bold200} color="primary">
                4.0
              </Text>
            </Flex>
            {categoryScreen ? (
              <Text fontSize="sm" className={styles.bold200} color="primary">
                S/. {post?.pst_precioBase}
              </Text>
            ) : (
              <ModalNewPost variant="primary" width="sm" icon />
            )}
          </Flex> */}
          <Flex align="center" justify="flex-start">
            <Text fontSize="md" className={styles.bold500} color="primary">
              {post?.pst_nombre}
            </Text>
          </Flex>
          {categoryScreen && (
            <Flex align="center" justify="flex-start">
              <Text fontSize="sm" className={styles.bold200} color="primary">
                Por {post.pstUsuarioId.us_nombre}
                {post.pstUsuarioId.us_apellido}
              </Text>
            </Flex>
          )}
        </Flex>
      </Box>
    </Box>
  )
}
