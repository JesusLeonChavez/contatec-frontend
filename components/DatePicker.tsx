import React, { useState } from "react"
import DatePicker from "react-datepicker"
import { registerLocale, setDefaultLocale } from  "react-datepicker";
import es from 'date-fns/locale/es';
import "react-datepicker/dist/react-datepicker.css"
import styles from "../styles/components/DatePicker.module.css"
registerLocale('es', es)

export default function DateP() {
  const [startDate, setStartDate] = useState(null)
  return (
    <div>
      <DatePicker
        className={styles.inputDate}
        locale="es"
        placeholderText="Seleccionar fecha"
        selected={startDate}
        dateFormat={"dd/MM/yyyy"}
        showMonthDropdown
        showYearDropdown
        dropdownMode="select"
        onChange={date => setStartDate(date)}
        minDate={new Date()}
        showDisabledMonthNavigation
        isClearable={true}
      />
    </div>
  )
}
