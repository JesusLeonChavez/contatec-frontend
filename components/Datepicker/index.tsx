import { Flex } from "@chakra-ui/react"
import React from "react"
import moment from "moment"
import ZIcon from "../Icon"
import Calendar from "./Subcomponents/Calendar"
import useOut from "./useOut"
import { PropsDatePicker } from "./type"
import styles from "./../../styles/components/Datepicker.module.css"

export default function DatePicker({
  fullWidth = true,
  disabled = false,
  onChange,
  value,
  formatLabel = "DD/MM/YYYY",
  position = "left",
  listPosition = "bottom",
  minDate = null,
  maxDate = null
}: PropsDatePicker) {
  const [date, setDate] = React.useState(value)
  const viewDate = date ? moment(new Date(date.toString())) : moment()
  const { ref, isOpen, setOpen } = useOut([])
  React.useEffect(() => {
    setDate(value)
  }, [value])
  function handleChange(val) {
    if (!value) {
      setDate(val)
    }
    onChange(val)
    setOpen(false)
  }

  function handleClick() {
    if (!isOpen) {
      setOpen(true)
    }
  }

  function handleFocus() {
    setOpen(true)
  }
  return (
    <Flex
      h="2.5rem"
      position="relative"
      bg="transparent"
      alignItems="center"
      border="1px solid"
      borderColor="gray.200"
      justifyContent="center"
      w={`${fullWidth ? "20rem" : "100%"}`}
      pointerEventer={`${disabled} ? 'aplique' : 'no-aplique'}`}
      onClick={handleClick}
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      ref={ref}
    >
      <input
        className={`${styles.inputDate} ${fullWidth && "100%"}`}
        onFocus={handleFocus}
        readOnly
        placeholder="Seleccione una fecha"
        value={date ? date.format(formatLabel) : ""}
      />
      <ZIcon className="icon" name="calendar" size={16} />
      <Flex
        w="250px"
        h="300px"
        // bg="red"
        borderWidth="1px"
        borderStyle="solid"
        borderColor="gray.400"
        position="absolute"
        bg="white"
        flexDirection="column"
        opacity="1"
        className={`${styles.date}
        ${listPosition === "top" ? styles.dP3 : ""}
        ${listPosition === "bottom" ? styles.dP4 : ""}
        ${isOpen ? styles.date2 : styles.date1}`}
      >
        <Calendar
          date={date}
          viewDate={viewDate}
          setDate={handleChange}
          minDate={minDate}
          maxDate={maxDate}
        />
      </Flex>
    </Flex>
  )
}
