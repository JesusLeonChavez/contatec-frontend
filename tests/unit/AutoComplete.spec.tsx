/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"
import "@testing-library/jest-dom"
import { newTheme } from "../../styles/theme"
import { ChakraProvider } from "@chakra-ui/react"
import { Autocomplete } from "../../components/AutoComplete"
// eslint-disable-next-line @typescript-eslint/no-empty-function
const props = { hits: [], currentRefinement: {}, refine: () => {} }
const ChakraRenderer = ({ children }) => {
  return <ChakraProvider theme={newTheme}>{children}</ChakraProvider>
}
describe("Autocomplete", function () {
  test("debe renderizarse", () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const component = render(<Autocomplete />, { wrapper: ChakraRenderer })

    expect(component).toMatchSnapshot()
  })

  test("debe pasar parametros", () => {
    const component = render(<Autocomplete {...props} />, {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      wrapper: ChakraRenderer
    })

    expect(component).toMatchSnapshot()
  })
})
