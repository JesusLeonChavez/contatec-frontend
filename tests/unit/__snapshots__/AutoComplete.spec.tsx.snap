// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete debe pasar parametros 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <form>
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-10m1y3x"
            for="text"
            id="text-label"
          >
            Busca el tema de tu interés
          </label>
          <div
            class="css-k008qs"
          >
            <ul
              style="text-decoration: none; list-style: none;"
            >
              <li>
                <div
                  class="chakra-input__group css-1ygfgic"
                >
                  <div
                    class="chakra-input__left-element css-1cw84h2"
                  >
                    <svg
                      color="var(--primary)"
                      fill="none"
                      height="1em"
                      style="font-size: 15px;"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.47 9.88h.66l4.157 4.166-1.241 1.241-4.167-4.158v-.658l-.225-.234a5.392 5.392 0 01-3.525 1.309 5.416 5.416 0 115.417-5.417 5.392 5.392 0 01-1.309 3.525l.234.225zM2.38 6.13a3.745 3.745 0 003.75 3.75 3.745 3.745 0 003.75-3.75 3.745 3.745 0 00-3.75-3.75 3.745 3.745 0 00-3.75 3.75z"
                        fill="var(--primary)"
                        fill-opacity="0.54"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  <input
                    autocomplete="off"
                    class="chakra-input css-aqiwep"
                    id="text"
                    name="search"
                    placeholder="P. ej: Desarrollo web"
                    type="search"
                    value="[object Object]"
                  />
                </div>
              </li>
            </ul>
            <button
              class="chakra-button css-xvfn2m"
              type="submit"
            >
              Buscar
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="chakra-form-control css-1kxonj9"
        role="group"
      >
        <label
          class="chakra-form__label css-10m1y3x"
          for="text"
          id="text-label"
        >
          Busca el tema de tu interés
        </label>
        <div
          class="css-k008qs"
        >
          <ul
            style="text-decoration: none; list-style: none;"
          >
            <li>
              <div
                class="chakra-input__group css-1ygfgic"
              >
                <div
                  class="chakra-input__left-element css-1cw84h2"
                >
                  <svg
                    color="var(--primary)"
                    fill="none"
                    height="1em"
                    style="font-size: 15px;"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.47 9.88h.66l4.157 4.166-1.241 1.241-4.167-4.158v-.658l-.225-.234a5.392 5.392 0 01-3.525 1.309 5.416 5.416 0 115.417-5.417 5.392 5.392 0 01-1.309 3.525l.234.225zM2.38 6.13a3.745 3.745 0 003.75 3.75 3.745 3.745 0 003.75-3.75 3.745 3.745 0 00-3.75-3.75 3.745 3.745 0 00-3.75 3.75z"
                      fill="var(--primary)"
                      fill-opacity="0.54"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <input
                  autocomplete="off"
                  class="chakra-input css-aqiwep"
                  id="text"
                  name="search"
                  placeholder="P. ej: Desarrollo web"
                  type="search"
                  value="[object Object]"
                />
              </div>
            </li>
          </ul>
          <button
            class="chakra-button css-xvfn2m"
            type="submit"
          >
            Buscar
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Autocomplete debe renderizarse 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <form>
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-10m1y3x"
            for="text"
            id="text-label"
          >
            Busca el tema de tu interés
          </label>
          <div
            class="css-k008qs"
          >
            <ul
              style="text-decoration: none; list-style: none;"
            >
              <li>
                <div
                  class="chakra-input__group css-1ygfgic"
                >
                  <div
                    class="chakra-input__left-element css-1cw84h2"
                  >
                    <svg
                      color="var(--primary)"
                      fill="none"
                      height="1em"
                      style="font-size: 15px;"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.47 9.88h.66l4.157 4.166-1.241 1.241-4.167-4.158v-.658l-.225-.234a5.392 5.392 0 01-3.525 1.309 5.416 5.416 0 115.417-5.417 5.392 5.392 0 01-1.309 3.525l.234.225zM2.38 6.13a3.745 3.745 0 003.75 3.75 3.745 3.745 0 003.75-3.75 3.745 3.745 0 00-3.75-3.75 3.745 3.745 0 00-3.75 3.75z"
                        fill="var(--primary)"
                        fill-opacity="0.54"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  <input
                    autocomplete="off"
                    class="chakra-input css-aqiwep"
                    id="text"
                    name="search"
                    placeholder="P. ej: Desarrollo web"
                    type="search"
                    value=""
                  />
                </div>
              </li>
            </ul>
            <button
              class="chakra-button css-xvfn2m"
              type="submit"
            >
              Buscar
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="chakra-form-control css-1kxonj9"
        role="group"
      >
        <label
          class="chakra-form__label css-10m1y3x"
          for="text"
          id="text-label"
        >
          Busca el tema de tu interés
        </label>
        <div
          class="css-k008qs"
        >
          <ul
            style="text-decoration: none; list-style: none;"
          >
            <li>
              <div
                class="chakra-input__group css-1ygfgic"
              >
                <div
                  class="chakra-input__left-element css-1cw84h2"
                >
                  <svg
                    color="var(--primary)"
                    fill="none"
                    height="1em"
                    style="font-size: 15px;"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.47 9.88h.66l4.157 4.166-1.241 1.241-4.167-4.158v-.658l-.225-.234a5.392 5.392 0 01-3.525 1.309 5.416 5.416 0 115.417-5.417 5.392 5.392 0 01-1.309 3.525l.234.225zM2.38 6.13a3.745 3.745 0 003.75 3.75 3.745 3.745 0 003.75-3.75 3.745 3.745 0 00-3.75-3.75 3.745 3.745 0 00-3.75 3.75z"
                      fill="var(--primary)"
                      fill-opacity="0.54"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <input
                  autocomplete="off"
                  class="chakra-input css-aqiwep"
                  id="text"
                  name="search"
                  placeholder="P. ej: Desarrollo web"
                  type="search"
                  value=""
                />
              </div>
            </li>
          </ul>
          <button
            class="chakra-button css-xvfn2m"
            type="submit"
          >
            Buscar
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
